test/primitive/phase_gate.jl

```diff
@@ -27,6 +27,11 @@ end
     @test (PhaseGate(2.0) == PhaseGate(2.0)) == true
 end
 
+@testset "test dispatch" begin
+    @test dispatch!(phase(0.1), 0.3) == phase(0.3)
```

Comment:
`dispatch!` is not designed to work with a `number`. One can use `setiparam!` instead.

Anyway, I am ok with dispatching a number.

Reply:
Yeah, but `Numer` is an iterator, so this should be allowed. And `setiparams!` is pretty low level.